--- annobin-12.93.orig/gcc-plugin/annobin.cc	2025-06-10 10:06:03.168778925 +0100
+++ annobin-12.93/gcc-plugin/annobin.cc	2025-06-10 10:59:10.724745093 +0100
@@ -2456,11 +2456,18 @@ annobin_active_check (const char * messa
   if (ends_with (annobin_input_filename, ".f"))
     return;
 
+#if 0 /* RHEL-95704 : GCC 14's version of warning() is different from GCC 15's.
+	 Whilst bootstrapping GTS-15 we build with GCC 14 but want to run with GGC 15.
+	 So we cannot safely use the warning() function.  */
+
   if (annobin_active_checks == 1)
     // FIXME: We should find an OPT_ value to use here so
     // that users can disable these warnings if they need to.
     warning (0, "%s", message);
   else if (annobin_active_checks == 2)
+#else
+  if (annobin_active_checks >= 1)
+#endif
     error ("%s", message);
   else
     ice ("unexpected value for annobin_active_checks");
--- annobin-12.93.orig/tests/active-checks-test	2025-06-10 10:06:03.169970481 +0100
+++ annobin-12.93/tests/active-checks-test	2025-06-10 11:13:27.075942964 +0100
@@ -20,7 +20,7 @@ GCC_OPTS="-D__FORTIFY_SOURCE=2 -DGLIBCXX
 
 start_test
 $GCC -fplugin=$PLUGIN $PLUGIN_OPTS -c $srcdir/hello.c $GCC_OPTS > gcc.out 2>&1
-if [ $? != 0 ];
+if [ $? != 1 ];
 then
     echo "$TEST_NAME: FAIL: could not compile test file"
     echo "$TEST_NAME: gcc command line: $GCC -fplugin=$PLUGIN $PLUGIN_OPTS -c $srcdir/hello.c $GCC_OPTS"
